lib/engine/stock_market.rb

```diff
@@ -12,11 +12,18 @@ def initialize(market, unlimited_types, multiple_buy_types: [], zigzag: nil)
       @zigzag = zigzag
       @market = market.map.with_index do |row, r_index|
         row.map.with_index do |code, c_index|
-          price = SharePrice.from_code(code,
-                                       r_index,
-                                       c_index,
-                                       unlimited_types,
-                                       multiple_buy_types: multiple_buy_types)
+          price = if code.instance_of?(Hash)
+                    SharePrice.new([r_index, c_index],
```

Comment:
SharePrice.new([r_index, c_index], **code)

Reply:
I believe ** is the hash expansion. It is for python for sure and I think that's the universal syntax. If not, there *is* a syntax for it where you don't have to specify each by hand.

Reply:
Changed